gui/RespiraConnectedDevice.h

```diff
@@ -0,0 +1,115 @@
+#include "../common/generated_libs/network_protocol/network_protocol.pb.h"
+#include "../common/third_party/nanopb/pb_common.h"
+#include "../common/third_party/nanopb/pb_decode.h"
+#include "../common/third_party/nanopb/pb_encode.h"
+#include "connected_device.h"
+#include <QSerialPort>
+#include <QThread>
+
+#define INTER_FRAME_TIMEOUT 11
+
+class RespiraConnectedDevice : public ConnectedDevice {
+
+public:
+  RespiraConnectedDevice(QString portName, uint32_t write_timeout_ms,
+                         uint32_t interframe_timeout_ms)
+      : serialPortName(portName), m_writeTimeout_ms(write_timeout_ms),
+        m_interframeTimeout_ms(interframe_timeout_ms) {}
+
+  ~RespiraConnectedDevice() {
+    if (m_serial != nullptr) {
+      m_serial->close();
+    }
+  }
+  void createPort() {
+    if (m_serial != nullptr)
+      return;
+    printf("Socket create\n");
+    m_serial = new QSerialPort();
+    m_serial->setPortName(serialPortName);
+    m_serial->setBaudRate(QSerialPort::Baud115200);
+    m_serial->setDataBits(QSerialPort::Data8);
+    m_serial->setParity(QSerialPort::NoParity);
+    m_serial->setStopBits(QSerialPort::OneStop);
+    m_serial->setFlowControl(QSerialPort::NoFlowControl);
+    if (!m_serial->open(QIODevice::ReadWrite)) {
+      printf("Could not open serial port %s\n",
+             serialPortName.toStdString().c_str());
+    }
+  }
+
+  bool SendGuiStatus(const GuiStatus &gui_status) override {
+    createPort();
+    uint8_t tx_buffer[GuiStatus_size];
+    pb_ostream_t stream = pb_ostream_from_buffer(tx_buffer, sizeof(tx_buffer));
+    if (!pb_encode(&stream, GuiStatus_fields, &gui_status)) {
+      // TODO: Serialization failure; log an error or raise an alert.
+      return false;
+    }
+
+    m_serial->write((const char *)tx_buffer, stream.bytes_written);
+
+    if (!m_serial->waitForBytesWritten(m_writeTimeout_ms)) {
+      // TODO communication failure, port closed? Log an error and raise
+      // an alert
+      return false;
+    }
+    return true;
+  }
+
+#include <chrono>
+  using SteadyClock = std::chrono::steady_clock;
+
+  bool ReceiveControllerStatus(ControllerStatus *controller_status) override {
+    createPort();
+    // Drop everything we have in buffer right now as we are not sure if this is
+    // a full frame or a part of it.
+    // We drop everything that comes in until we get at least half of interframe
+    // silence
+
+    //    if (m_serial->bytesAvailable() > 0) {
```

Comment:
Remove commented-out code

Reply:
Yep. I still don't like the fact that we potentially try to deserialize some garbage, but having this here and now creates too much packet timeouts. This will be non-issue when framing comes along.